src/App.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import './App.css';
+import { Route, Routes } from 'react-router-dom';
+import Navigation from './components/navigation';
+import BooksDisplay from './components/booksDisplay';
+import Categories from './components/categories';
+
+function App() {
+  return (
+    <div className="wholeApp">
+      <Navigation />
+      <Routes>
+        <Route path="/" element={<BooksDisplay />} />
+        <Route path="/categories" element={<Categories />} />
+        <Route path="*" element={<BooksDisplay />} />
```

Comment:
- **[OPTIONAL]** The "\*" path in the given code is necessary as a catch-all route. It is used when none of the specified routes match the current URL. In this case, if the user enters a URL that doesn't match "/" or "/categories", the "\*" route will be used and the <BooksDisplay /> component will be rendered. It could be better to take the user to a "NOT FOUND" page for a good UX.

Reply:
thankyou, @franclobo for the great review 👍. i will implement the optional suggestion as well.